### PR TITLE
[HWORKS-1048] Support multiple modularized project environments

### PR DESCRIPTION
**Pull request by robzor92**

### CODE REVIEW

File: python/hsml/deployment.py [hunk 1]
```diff
@@ -461,6 +461,15 @@ def api_protocol(self):
     def api_protocol(self, api_protocol: str):
         self._predictor.api_protocol = api_protocol
 
+    @property
+    def environment(self):
+        """Name of inference environment"""
```
Comment (javierdlrm) [inline python/hsml/deployment.py:466]: Maybe this?

```suggestion
        """Name of model serving environment"""
```

File: python/hsml/predictor.py [hunk 2]
```diff
@@ -268,6 +270,9 @@ def extract_fields_from_json(cls, json_decamelized):
         kwargs["created_at"] = json_decamelized.pop("created")
         kwargs["creator"] = json_decamelized.pop("creator")
         kwargs["api_protocol"] = json_decamelized.pop("api_protocol")
+        if "environmentdto" in json_decamelized:
```
Comment (javierdlrm) [inline python/hsml/predictor.py:273]: Doesn't `environmentDTO` get mapped to `environment_dto`?

File: python/tests/test_model.py [hunk 3]
```diff
@@ -202,7 +202,7 @@ def test_deploy(self, mocker, backend_fixtures):
             inference_logger=inference_logger,
             inference_batcher=inference_batcher,
             transformer=transformer,
-            api_protocol=p_json["api_protocol"],
```
Comment (javierdlrm) [inline python/tests/test_model.py:205]: Why skipping api_protocol?

File: python/tests/fixtures/predictor_fixtures.json [hunk 4]
```diff
@@ -40,6 +40,9 @@
           "inference_logging": "ALL",
           "kafka_topic_dto": {
             "name": "topic"
+          },
+          "environment_dto": {
```
Comment (javierdlrm) [inline python/tests/fixtures/predictor_fixtures.json:44]: Can you add asserts for `environment_dto` in the [predictor tests](https://github.com/logicalclocks/machine-learning-api/blob/954a07b9b2499f485e0d4678bb87b6c3e260a123/python/tests/test_predictor.py#L79)?

**Review by javierdlrm — CHANGES_REQUESTED** [review]

File: python/tests/test_model.py (hunk 3 above)
Comment (robzor92) [reply python/tests/test_model.py:205]: mistake, I reverted

File: python/tests/fixtures/predictor_fixtures.json (hunk 4 above)
Comment (robzor92) [reply python/tests/fixtures/predictor_fixtures.json:44]: added many test cases here

**Review by javierdlrm — APPROVED** [review]

**Pull request merged**